pkg/cmd/infra/router/template.go

```diff
@@ -421,7 +420,7 @@ func (o *TemplateRouterOptions) Run() error {
 		nextPlugin = controller.NewExtendedValidator(nextPlugin, controller.RejectionRecorder(statusPlugin))
 	}
 	uniqueHostPlugin := controller.NewUniqueHost(nextPlugin, o.RouteSelectionFunc(), o.RouterSelection.DisableNamespaceOwnershipCheck, controller.RejectionRecorder(statusPlugin))
-	plugin := controller.NewHostAdmitter(uniqueHostPlugin, o.RouteAdmissionFunc(), o.AllowWildcardRoutes, o.RouterSelection.DisableNamespaceOwnershipCheck, controller.RejectionRecorder(statusPlugin))
+	plugin := controller.NewHostAdmitter(uniqueHostPlugin, o.RouteAdmissionFunc(), o.AllowWildcardRoutes, o.RouterSelection.DisableNamespaceOwnershipCheck, controller.RejectionRecorder(statusPlugin), o.NamespaceLabels)
```

Comment:
NamespaceLabels is only one of the methods to do router sharding.
I think you need to check:
```
sharding := flase
if (NamespaceLabels != nil) || (ProjectLabels != nil) || (len(Namespace) > 0) {
   sharding = true
}
```
and in https://github.com/openshift/origin/pull/18511/files#diff-0df751f9a9e4d626778b1c3ec7b68b18R128
you could do:
```
if err := p.admitter(route); (err != nil) && sharding {
...
```

Reply:
thanks, i'll add that